### PR TITLE
Generalised fonts

### PR DESCRIPTION
**Pull request by Lignum**
As @Lemmmy requested, this generalises fonts, so that each renderer may decide how to load and render them. This still needs to be thoroughly tested!

### CODE REVIEW

File: src/main/java/net/clgd/ccemux/rendering/tror/TRoRRenderer.java [hunk 1]
```diff
@@ -48,6 +49,11 @@ public void removeListener(Renderer.Listener listener) {
 	}
 
 	@Override
+	public TerminalFont loadFont(InputStream stream) throws IOException {
+		return null;
```
Comment (dmarcuse) [inline src/main/java/net/clgd/ccemux/rendering/tror/TRoRRenderer.java]: Would probably be better to return a "no-op" font object rather than `null`

**Review by dmarcuse — APPROVED** [review]

File: src/main/java/net/clgd/ccemux/rendering/TerminalFonts.java [hunk 2]
```diff
@@ -85,11 +72,42 @@ private void registerFont(TerminalFont font) {
 	 *
 	 * @return The "best" font
 	 */
-	public TerminalFont getBest() {
-		val fonts = explicitFonts.size() > 0 ? explicitFonts : implicitFonts;
-		return fonts.stream()
-				.sorted(Comparator.comparingInt(a -> a.getCharWidth() + a.getCharHeight()))
-				.reduce((a, b) -> b)
-				.orElseThrow(() -> new RuntimeException("No terminal fonts available"));
+	public TerminalFont getBest(Renderer renderer) {
```
Comment (dmarcuse) [inline src/main/java/net/clgd/ccemux/rendering/TerminalFonts.java]: `@Getter(lazy=true)` would simplify this

**Review by Lemmmy — APPROVED** [review]

**Comment by Lemmmy** [comment]
can we change renderer back to an interface befroe merging

```java
public class GDXAdapter extends ApplicationAdapter implements Renderer {
```

otherwise this is a pain in arse

**Review by Lemmmy — CHANGES_REQUESTED** [review]
iabstract casss nargvrgvefds

**Review by Lemmmy — APPROVED** [review]

**Pull request merged**